l1-contracts/scripts/upgrade-shared-bridge-era.ts

```diff
@@ -0,0 +1,76 @@
+// hardhat import should be the first import in the file
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import * as hardhat from "hardhat";
+import { Command } from "commander";
+import { Wallet, ethers } from "ethers";
+import { Deployer } from "../src.ts/deploy";
+import { formatUnits, parseUnits, Interface } from "ethers/lib/utils";
+import { web3Provider, GAS_MULTIPLIER } from "./utils";
+import { deployedAddressesFromEnv } from "../src.ts/deploy-utils";
+import { ethTestConfig } from "../src.ts/utils";
+
+const provider = web3Provider();
+
+async function main() {
+  const program = new Command();
+
+  program.version("0.1.0").name("upgrade-shared-bridge-era").description("upgrade shared bridge for era diamond proxy");
+
+  program
+    .option("--private-key <private-key>")
+    .option("--chain-id <chain-id>")
+    .option("--gas-price <gas-price>")
+    .option("--nonce <nonce>")
+    .option("--owner-address <owner-address>")
+    .option("--create2-salt <create2-salt>")
+    .option("--diamond-upgrade-init <version>")
+    .option("--only-verifier")
+    .action(async (cmd) => {
+      const deployWallet = cmd.privateKey
+        ? new Wallet(cmd.privateKey, provider)
+        : Wallet.fromMnemonic(
+            process.env.MNEMONIC ? process.env.MNEMONIC : ethTestConfig.mnemonic,
+            "m/44'/60'/0'/0/1"
+          ).connect(provider);
+      console.log(`Using deployer wallet: ${deployWallet.address}`);
+
+      const ownerAddress = cmd.ownerAddress ? cmd.ownerAddress : deployWallet.address;
+      console.log(`Using owner address: ${ownerAddress}`);
+
+      const gasPrice = cmd.gasPrice
+        ? parseUnits(cmd.gasPrice, "gwei")
+        : (await provider.getGasPrice()).mul(GAS_MULTIPLIER);
+      console.log(`Using gas price: ${formatUnits(gasPrice, "gwei")} gwei`);
+
+      const nonce = cmd.nonce ? parseInt(cmd.nonce) : await deployWallet.getTransactionCount();
+      console.log(`Using nonce: ${nonce}`);
+
+      const create2Salt = cmd.create2Salt ? cmd.create2Salt : ethers.utils.hexlify(ethers.utils.randomBytes(32));
+
+      const deployer = new Deployer({
+        deployWallet,
+        addresses: deployedAddressesFromEnv(),
+        ownerAddress,
+        verbose: true,
+      });
+
+      await deployer.deploySharedBridgeImplementation(create2Salt, { nonce });
+
+      const proxyAdminInterface = new Interface(hardhat.artifacts.readArtifactSync("ProxyAdmin").abi);
+      const calldata = proxyAdminInterface.encodeFunctionData("upgrade(address,address)", [
```

Comment:
Same here, for the non-localhost upgrades we should manually generate calldata for the upgrades coming from governance, unless we expect that this particular upgrade will be always done by an EOA